patroni/ha.py

```diff
@@ -491,7 +526,10 @@ def reinitialize(self, cluster):
 
     def process_scheduled_action(self):
         if self.reinitialize_scheduled():
-            if self.cluster.is_unlocked():
+            if self.is_paused():
+                logger.warning('Cluster is in a pause state, can not reinitialize')
```

Comment:
I think we ran reinitialize as a scheduled action in order to avoid running it at an arbitrary time, where it might interfere with out checks (i.e. recover will try to run a stopped PostgreSQL). On the other hand, it seems to be the only place and we can easily introduce a flag to avoid it doing so. The question is whether the reinitialize thread spawned from the short-lived API thread will block the latter from termination (probably not, we can check it).

Anyway, reinitialize is only scheduled for technical reasons, not because the user wanted it to be scheduled at a certain time, so I don't thing it's a strong reason for blocking it during pause (we already do bootstrap during pause, which is also a 'scheduled' action) .